src/environmentd/src/lib.rs

```diff
@@ -308,21 +311,59 @@ impl Listeners {
         let active_connection_counter = ConnectionCounter::default();
         let (deployment_state, deployment_state_handle) = DeploymentState::new();
 
+        let webhook_concurrency_limit = WebhookConcurrencyLimiter::default();
+        let internal_route_config = InternalRouteConfig {
+            deployment_state_handle,
+            internal_console_redirect_url: config.internal_console_redirect_url,
+        };
+
+        let (authenticator_frontegg_tx, authenticator_frontegg_rx) = oneshot::channel();
+        let authenticator_frontegg_rx = authenticator_frontegg_rx.shared();
+        let (authenticator_none_tx, authenticator_none_rx) = oneshot::channel();
+        let authenticator_none_rx = authenticator_none_rx.shared();
```

Comment:
This is partially in preparation for `Authenticator::Password(AdapterClient)` support coming in the next PR. While we can immediately create the `Authenticator::Frontegg` and `Authenticator::None` now, we won't be able to immediately create `Authenticator::Password`, since we want to launch this HTTP server before the `AdapterClient` is instantiated.

Reply:
We can't construct an `Authenticator::Password` because it contains an `AdapterClient`, which isn't ready yet. We want the listener to be available for metrics endpoints during the startup, so we can't wait to start the HTTP server until after the `AdapterClient` is ready.

Reply:
Gotcha. I remember having to do something similar with pgwire. Perhaps in the future we should split out the metrics server if it's lifecycle is separate.